packages/kbn-securitysolution-ecs/src/index.ts

```diff
@@ -1,80 +1,153 @@
+ * Side Public License, v 1.
  */
 
-import type { AgentEcs } from './agent';
+import type { Ecs } from '@kbn/ecs';
+import { AgentEcs } from './agent';
 import type { AuditdEcs } from './auditd';
+import type { CloudEcs } from './cloud';
+import type { CodeSignature, FileEcs } from './file';
 import type { DestinationEcs } from './destination';
-import type { DnsEcs } from './dns';
 import type { DllEcs } from './dll';
+import type { DnsEcs } from './dns';
 import type { EndgameEcs } from './endgame';
+import { EventCategory, EventCode } from './event';
 import type { EventEcs } from './event';
-import type { FileEcs } from './file';
 import type { GeoEcs } from './geo';
-import type { HostEcs } from './host';
+import type { HostEcs, OsEcs } from './host';
+import type { HttpEcs } from './http';
+import type { MemoryProtection } from './memory_protection';
 import type { NetworkEcs } from './network';
+import type { ProcessEcs } from './process';
+import type { Ransomware } from './ransomware';
 import type { RegistryEcs } from './registry';
 import type { RuleEcs } from './rule';
 import type { SignalEcs, SignalEcsAAD } from './signal';
 import type { SourceEcs } from './source';
 import type { SuricataEcs } from './suricata';
+import type { SystemEcs } from './system';
+import type { Target } from './target_type';
+import type { ThreatEcs, ThreatIndicatorEcs } from './threat';
 import type { TlsEcs } from './tls';
-import type { ZeekEcs } from './zeek';
-import type { HttpEcs } from './http';
 import type { UrlEcs } from './url';
 import type { UserEcs } from './user';
 import type { WinlogEcs } from './winlog';
-import type { ProcessEcs } from './process';
-import type { SystemEcs } from './system';
-import type { ThreatEcs } from './threat';
-import type { Ransomware } from './ransomware';
-import type { MemoryProtection } from './memory_protection';
-import type { Target } from './target_type';
+import type { ZeekEcs } from './zeek';
+export * from './ecs_fields';
 
-export interface Ecs {
-  _id: string;
-  _index?: string;
+export { EventCategory, EventCode };
+
+export type {
+  AgentEcs,
+  AuditdEcs,
+  CloudEcs,
+  CodeSignature,
+  DestinationEcs,
+  DllEcs,
+  DnsEcs,
+  EndgameEcs,
+  EventEcs,
+  FileEcs,
+  GeoEcs,
+  HostEcs,
+  HttpEcs,
+  MemoryProtection,
+  NetworkEcs,
+  OsEcs,
+  ProcessEcs,
+  Ransomware,
+  RegistryEcs,
+  RuleEcs,
+  SignalEcs,
+  SourceEcs,
+  SuricataEcs,
+  SystemEcs,
+  Target,
+  ThreatEcs,
+  ThreatIndicatorEcs,
+  TlsEcs,
+  UrlEcs,
+  UserEcs,
+  WinlogEcs,
+  ZeekEcs,
+};
+
+type OverrideTypes =
+  | 'agent'
+  | 'destination'
+  | 'dns'
+  | 'ecs'
+  | 'event'
+  | 'file'
+  | 'host'
+  | 'http'
+  | 'message'
+  | 'network'
+  | 'process'
+  | 'registry'
+  | 'rule'
+  | 'source'
+  | 'threat'
+  | '@timestamp'
+  | 'tls'
+  | 'url'
+  | 'user';
+
+// Security Solution Extension of the Elastic Common Schema
+export interface EcsSecurityExtension extends Omit<Ecs, OverrideTypes> {
```

Comment:
@andrew-goldstein @YulNaumenko I'm thinking for this PR, I am not even going to extend since we're not actually using any of the Ecs yet. I will extend in [part 2](https://github.com/elastic/kibana/issues/148676) when we're actually in line with the Ecs
```suggestion
export interface EcsSecurityExtension {
```